spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -0,0 +1,78 @@
+# typed: ignore
+
+require 'datadog/profiling/spec_helper'
+require 'datadog/profiling/stack_recorder'
+
+RSpec.describe Datadog::Profiling::StackRecorder do
+  before { skip_if_profiling_not_supported(self) }
+
+  subject(:stack_recorder) { described_class.new }
+
+  # NOTE: A lot of libddprof integration behaviors are tested in the Collectors::Stack specs, since we need actual
+  # samples in order to observe what comes out of libddprof
+
+  describe '#serialize' do
+    subject(:serialize) { stack_recorder.serialize }
+
+    let(:start) { serialize[0] }
+    let(:finish) { serialize[1] }
+    let(:encoded_pprof) { serialize[2] }
+
+    let(:decoded_profile) { ::Perftools::Profiles::Profile.decode(encoded_pprof) }
+
+    context 'when the profile is empty' do
```

Comment:
Note: Tests for when a profile is non-empty will be added later in #1966.

This is because the `StackRecorder` by itself doesn't generate samples, they are added by other components using `record_sample`. (And `record_sample` is only a C API, and I chose not to expose it to Ruby just for the tests here, since anyway in #1966 a better option comes along)